syntax = "proto3";

package StockTrading;

service StockTrading{
rpc Lookup(LookupRequest) returns (LookupResponse) {}
rpc Trade(TradeRequest) returns (TradeResponse) {}
rpc Update(UpdateRequest) returns (UpdateResponse) {}

}


message LookupRequest {
  string stock_name = 1;
}

message LookupResponse {
  float price = 1;
  int32 volume = 2;
}

message TradeRequest {
  string stock_name = 1;
  int32 amount = 2;
  bool is_buy = 3;
}

message TradeResponse {
  int32 status = 1;
}

message UpdateRequest {
  string stock_name = 1;
  float price = 2;
}

message UpdateResponse {
  int32 status = 1;
}


